app/composables/useDataHover.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * Composable for managing hover state across cache analysis data elements
+ */
+import { formatDuration, intervalToDuration } from 'date-fns'
+
+interface HoverState {
+  dataKey: string | null
+  dataValue: string | null
+  rawValue: unknown | null
+}
+
+const hoverState = ref<HoverState>({
+  dataKey: null,
+  dataValue: null,
+  rawValue: null,
+})
+
+export const useDataHover = () => {
+  const setHover = (dataKey: string | null, dataValue: string | null = null, rawValue: unknown = null) => {
+    hoverState.value = { dataKey, dataValue, rawValue }
+  }
+
+  const clearHover = () => {
+    hoverState.value = { dataKey: null, dataValue: null, rawValue: null }
+  }
+
+  const isKeyHovered = (key: string) => {
+    return hoverState.value.dataKey === key
+  }
+
+  const isValueMatching = (rawValue: unknown) => {
+    const hoveredRawValue = hoverState.value.rawValue
+
+    // Handle Date comparison specially
+    if (hoveredRawValue instanceof Date && rawValue instanceof Date) {
+      return hoveredRawValue.getTime() === rawValue.getTime()
+    }
+
+    // For all other types, use direct equality
+    return hoveredRawValue === rawValue
+  }
+
+  const getDelta = (currentRawValue: unknown): string | null => {
+    const hoveredRawValue = hoverState.value.rawValue
+
+    // Calculate delta for numbers
+    if (typeof hoveredRawValue === 'number' && typeof currentRawValue === 'number') {
+      const delta = currentRawValue - hoveredRawValue
+      if (delta === 0) return null
+
+      // Format delta with appropriate sign and unit
+      const sign = delta > 0 ? '+' : ''
+      return `${sign}${delta}s`
+    }
+
+    // Calculate delta for Dates
```

Comment:
Unfortunately I tested this and it isn't working :(. I don't see a delta for dates that are different.